backbone.js

```diff
@@ -890,15 +891,19 @@
       // Silently sort the collection if appropriate.
       if (sort) this.sort({silent: true});
 
-      // Unless silenced, it's time to fire all appropriate add/sort events.
+      // Unless silenced, it's time to fire all appropriate add/update/sort events.
       if (!options.silent) {
         var addOpts = at != null ? _.clone(options) : options;
         for (var i = 0; i < toAdd.length; i++) {
           if (at != null) addOpts.index = at + i;
           (model = toAdd[i]).trigger('add', model, this, addOpts);
         }
+        if (toAdd.length || toRemove.length) {
+          if (toAdd.length) options.addedModels = toAdd;
+          if (toRemove.length) options.previousModels = toRemove;
```

Comment:
why not just always set `addedModels` and `previousModels` on options and trigger updates only if theres a `length` change. Seems mostly harmless (actually better IMO as the user won't have to check if `toAdd` is an array or not)